ebos/eclgenericproblem.cc

```diff
@@ -374,54 +374,66 @@ updateNum(const std::string& name, std::vector<T>& numbers)
     unsigned numElems = gridView_.size(/*codim=*/0);
     numbers.resize(numElems);
     for (unsigned elemIdx = 0; elemIdx < numElems; ++elemIdx) {
-        numbers[elemIdx] = static_cast<T>(numData[elemIdx]) - 1;
+        if (numData[elemIdx] > num_regions) {
```

Comment:
Would you mind `static_cast<>`-ing `num_regions` into an `int`, please?  That way we don't get additional [signed vs. unsigned comparison](https://ci.opm-project.org/job/opm-simulators-PR-builder/5040/gcc/folder.3106625/source.93ca4d50-d5bf-4f86-adea-1c0ace1af6c9/#377) diagnostics.